src/page/SellHistory.tsx

```diff
@@ -0,0 +1,116 @@
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import { styled } from 'styled-components';
+import { getSellHistory } from '../api/SellHistoryFetcher';
+import { Badge, BadgeProps } from '../components/Badge';
+import { Error } from '../components/Error';
+import { Header } from '../components/Header';
+import { Loader } from '../components/Loader';
+import { ProductItem } from '../components/ProductItem';
+import { useAuthStore } from '../stores/useAuthStore';
+import { ItemBaseType } from '../types';
+
+type HistoryData = {
+  items: ItemBaseType[];
+  nextCursor: number;
+};
+
+export function SellHistory() {
+  const { nickname } = useAuthStore();
+  const [isSold, setIsSold] = useState<boolean>();
+  const {
+    data: historyData,
+    isLoading,
+    isError,
+  } = useQuery<HistoryData, Error>(['history', isSold], () =>
+    getSellHistory({ nickname, isSold })
+  );
+
+  // 무한 스크롤 구현
+  const setBadgeOption = (text: string, status: boolean | undefined) => {
```

Comment:
아래 50~52 라인에서 함수 호출하는 부분을 보면 '전체', '판매 중', '판매 완료'가 상태를 나타내는 건지 뱃지에 표현되는 텍스트인지 알 수가 없고 true, false도 무엇에 대한 값인지 알 수가 없네용!
객체 형식의 파라미터를 사용해보면 개선이 될 것 같아요.

```suggestion
  const setBadgeOption = ({ text: string, isSelected: booloean }) => {
```
boolean 값인데 undefined 까지 넣어줄 필요는 없어 보이구요!

Reply:
말씀해 주신 방법처럼 객체로 표현하니 매칭 된 값이 어떤 값인지 알 수 있어서 좋은 것 같네요👍